Source/Plugin.BLE.Abstractions/AdapterBase.cs

```diff
@@ -108,36 +108,51 @@ public async Task ConnectToDeviceAsync(IDevice device, ConnectParameters connect
 
             using (var cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken))
             {
```

Comment:
Wrapped it all in a while-loop. So we keep retrying until cancellation or a connection